### PR TITLE
Minor rewrite of env::current_exe docs; clarify symlinks.

### PR DESCRIPTION
**Pull request by frewsxcv**
- Update example in ‘security’ section to use hard links, like the
  linked securityvulns.com example.
- Weaken language on symbolic links – indicate behavior is
  platform-specific

Fixes https://github.com/rust-lang/rust/issues/43617.

### CODE REVIEW

File: src/libstd/env.rs
Comment (TimNN) [inline src/libstd/env.rs]: Was the change from `'` to `’` deliberate? (Not sure what the convention is here).

**Comment by TimNN** [comment]
Looks good to me apart from the comment, however I'm unfamiliar with the current style conventions around Rust docs, so 

r? @steveklabnik

File: src/libstd/env.rs
Comment (frewsxcv) [reply src/libstd/env.rs]: deliberate but not something i feel strongly about. figure the line is already changing so i'm not ruining the diff in any way, but i'm fine keeping as-is

Comment (TimNN) [reply src/libstd/env.rs]: I don't mind much either way however it looks like `'` is used a lot more than `’` in the existing doc comments, so my personal vote would be to keep `'` for consistency (but feel free to keep this as-is to get a second opinion).

Comment (nagisa) [reply src/libstd/env.rs]: `’` is the preferred symbol to use for apostrophe in english text according to the Unicode standard. `'` is more commonplace because people can’t always be bothered to type Unicode symbols.

My personal vote is to follow the preferences set forth by the standard (I myself try to type unicode `’`), but is not something I would bug other people about whether they do or don’t follow the practice.

**Comment by alexcrichton** [comment]
ping @steveklabnik (or @rust-lang/docs), mind taking a look?

**Comment by GuillaumeGomez** [comment]
Seems good for me.

File: src/libstd/env.rs
Comment (QuietMisdreavus) [reply src/libstd/env.rs]: In all the docs i've written, i've just used the regular `'` because `’` is impossible to type on US-layout keyboards without a software compose key or other kind of software text replacement. On the other hand, the difference is likely too small to notice unless it was inconsistent within a single page. So here it's probably fine, but i wouldn't go using it on other pages without grabbing all the others on the same page like you did here. (And then possibly creating a tidy lint for it so we can keep on top of it. `>_>`)

**Review by QuietMisdreavus — APPROVED** [review]

**Comment by QuietMisdreavus** [comment]
Looks good to me. The comment about `'` versus `’` is not worth holding up the docs update for.

@bors r+ rollup

**Pull request merged**
